### PR TITLE
Fixes XML Map Export

### LINKED ISSUE #2068: exporting a Map to XML causes a DOMException
The jme3-plugins sub-project provides an `XMLExporter` class to export savables to XML files.

However, if the `Savable` includes a non-empty `Map`, invoking the `save()` method causes `CoreDocumentImpl.createAttribute()` to throw a `DOMException`.

Test code:
```java
        Map<String, String> map = new HashMap<>();
        map.put("key", "value");
        rootNode.setUserData("map", map);

        String xmlPath = "TestExport.xml";
        File xmlFile = new File(xmlPath);
        JmeExporter exporter = XMLExporter.getInstance();
        try {
            exporter.save(rootNode, xmlFile);
        } catch (IOException exception) {
            logger.log(Level.SEVERE, exception.getMessage(), exception);
        }
```

Typical result:
```console
Aug 29, 2023 5:24:53 PM com.jme3.app.LegacyApplication handleError
SEVERE: Uncaught exception thrown in Thread[jME3 Main,5,main]
org.w3c.dom.DOMException: INVALID_CHARACTER_ERR: An invalid or illegal XML character is specified.
	at java.xml/com.sun.org.apache.xerces.internal.dom.CoreDocumentImpl.createAttribute(CoreDocumentImpl.java:573)
	at java.xml/com.sun.org.apache.xerces.internal.dom.ElementImpl.setAttribute(ElementImpl.java:503)
	at com.jme3.export.xml.DOMOutputCapsule.write(DOMOutputCapsule.java:165)
	at com.jme3.scene.UserData.writeList(UserData.java:258)
	at com.jme3.scene.UserData.write(UserData.java:172)
	at com.jme3.export.xml.DOMOutputCapsule.write(DOMOutputCapsule.java:555)
	at com.jme3.export.xml.DOMOutputCapsule.writeStringSavableMap(DOMOutputCapsule.java:728)
	at com.jme3.scene.Spatial.write(Spatial.java:1626)
	at com.jme3.scene.Node.write(Node.java:757)
	at com.jme3.export.xml.DOMOutputCapsule.write(DOMOutputCapsule.java:555)
	at com.jme3.export.xml.XMLExporter.save(XMLExporter.java:70)
	at com.jme3.export.xml.XMLExporter.save(XMLExporter.java:88)
	at com.jme3.export.JmeExporter.save(JmeExporter.java:62)
```

Note that the `BinaryExporter` class seems to handle maps just fine.

### PR DESCRIPTION
**Pull request by codex128**
Updates the J3o version from 2 to 3 for adding "jme-" prefix to userdata maps, lists, and arrays to fix #2068 xml export issue.

### CODE REVIEW

**Comment by riccardobl** [comment]
I think the prefix should be added and removed inside

jme3-plugins/src/xml/java/com/jme3/export/xml/DOMOutputCapsule.java
jme3-plugins/src/xml/java/com/jme3/export/xml/DOMInputCapsule.java

since it fixes a limitation that is specific to xml specs

**Comment by riccardobl** [comment]
Can you bump the format to 3 as you did in the first draft and enable the prefix only for formats >=3 ? So that we can keep backward compatibility for already exported xml.

**Comment by codex128** [comment]
Sorry, I should have thought of that before. :P

File: jme3-plugins/src/xml/java/com/jme3/export/xml/DOMInputCapsule.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public DOMInputCapsule(Document doc, XMLImporter importer) {
         this.importer = importer;
         currentElem = doc.getDocumentElement();
         
-        String version = currentElem.getAttribute("format_version");
+        String version = XMLUtils.getAttribute(currentElem, "format_version");
```
Comment (riccardobl) [inline jme3-plugins/src/xml/java/com/jme3/export/xml/DOMInputCapsule.java:75]: This should be always unprefixed for backward compatibility

File: jme3-plugins/src/xml/java/com/jme3/export/xml/XMLUtils.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+     * <p>
+     * Automatically appends {@link #PREFIX} to the beginning
+     * of the name before looking up the attribute for format versions 3 and up.
+     * 
+     * @param element XML element to get the attribute from
+     * @param name name of the attribute (without prefix)
+     * @return named attribute
+     */
+    public static String getAttribute(Element element, String name) {
+        if (FormatVersion.VERSION >= 3) {
```
Comment (riccardobl) [inline jme3-plugins/src/xml/java/com/jme3/export/xml/XMLUtils.java:78]: To support old files, the check should be against `importer.formatVersion`  that is the version of the imported file, and not `FormatVersion.VERSION` that is the current supported version.

**Review by riccardobl — CHANGES_REQUESTED** [review]

**Review by riccardobl — APPROVED** [review]

**Comment by riccardobl** [comment]
Looks fine now, thank you!

File: jme3-plugins/src/xml/java/com/jme3/export/xml/DOMOutputCapsule.java [hunk 3]
```diff
@@ -80,7 +80,7 @@ public Document getDoc() {
     private Element appendElement(String name) {
         Element ret = doc.createElement(name);
         if (currentElement == null) {
-            ret.setAttribute("format_version", Integer.toString(FormatVersion.VERSION));
+            XMLUtils.setAttribute(ret, "format_version", Integer.toString(FormatVersion.VERSION));
```
Comment (riccardobl) [inline jme3-plugins/src/xml/java/com/jme3/export/xml/DOMOutputCapsule.java:83]: This should not be prefixed, same as in the input capsule

**Review by riccardobl — CHANGES_REQUESTED** [review]

**Comment by riccardobl** [comment]
Sorry, i did one last review before the merge and i've found there is still one last issue, the format_version is prefixed on the output capsule

**Review by riccardobl — APPROVED** [review]

**Pull request merged**